src/main/java/org/openapitools/client/model/SemestersEnum.java

```diff
@@ -34,7 +34,7 @@ public static SemestersEnum fromValue(BigDecimal value) {
 
     public static SemestersEnum fromValue(String semesterId) {
         for (SemestersEnum b : SemestersEnum.values()) {
-            if (b.name().equals(semesterId)) {
+            if (b.valueStr.equals(semesterId)) {
```

Comment:
This will affect the `add` command since it uses the enum name as command parameters

Reply:
Take a look at the storage of `SavedModule`
https://github.com/AY2223S1-CS2103-F14-1/tp/blob/f8888d4422c04723fb511b560e650c15ec27596a/src/main/java/nus/climods/storage/module/user/JsonAdaptedUserModule.java#L35-L40


I changed the storage of `SavedModule` to use the enum name instead so that we will not need to deal with 2 ways of deserialization. One where the `add` command uses enum name and storage that uses the full description string

Reply:
you can try deleting your existing cache and lmk if the bug still persist

Reply:
works, i realised i needed to change my cached user module and the files i use for tests 🤡